obj/empty.go

```diff
@@ -21,22 +17,15 @@ func isEmptyValue(v reflect.Value) bool {
 		return v.Float() == 0
 	case reflect.Interface, reflect.Ptr:
 		return v.IsNil()
```

Comment:
Given the fact that we consider a length-0 slice/array to be "empty" (but non-nil), this check here is slightly odd. For example, an empty map stored in an interface will be considered non-empty (right?).

Reply:
I'm not sure I entirely parsed that question correctly, but perhaps the method name is odd and it should actually be called something involving the word "zero"...?

[this is fairly literally identical to the checks used in stdlib json](https://github.com/golang/go/blob/541f9c0345d4ec52d9f4be5913bf8097f687e819/src/encoding/json/encode.go#L314-L330), if that's helpful reference.

Reply:
Interesting. Basically, my point is that it's *slightly* looser than an "isZero" check. That is, we admit a non-nil zero-length slice while the zero value is technically `nil`.

Note: I think this is as correct as we can get, it just feels like there are some odd edge-cases.